### PR TITLE
feat: add hasVerifiedAttestations and getVerifiedAttestations

### PR DESCRIPTION
**Pull request by alvaroraminelli**
**What changed? Why?**

Introduce new functionalities that make easier verify addresses in the Ethereum Attestation Service (EAS).

`hasVerifiedAttestations`
`getAttestation`

**Notes to reviewers**

**How has it been tested?**

Local build and consuming in a playground

### CODE REVIEW

**Comment by alvaroraminelli** [comment]
- Missing tests

File: src/core/getAttestation.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      revocationTime
+      schemaId
+      timeCreated
+      txid
+    }
+  }
+`;
+
+/**
+ * Retrieves attestations for a given address and chain, optionally filtered by schemas.
```
[the remark this answers is not in the document]
Comment (alvaroraminelli-cb) [reply src/core/getAttestation.ts:28]: what is the path to support all attestations in general?

File: README.md [hunk 2]
```diff
@@ -403,6 +403,30 @@ type UseName = {
 };
 ```
 
+### hasVerifiedAttestations
+
+```tsx
+import { base } from 'viem/chains';
+import { hasVerifiedAttestations } from '@coinbase/onchainkit';
+
+const hasVerifiedAttestation = await hasVerifiedAttestations(
```
Comment (Zizzamia) [inline README.md:412]: Is `has` the right wording for an API 🤔

Also can we we document what't the return type.

File: README.md [hunk 3]
```diff
@@ -403,6 +403,30 @@ type UseName = {
 };
 ```
 
+### hasVerifiedAttestations
+
```
Comment (Zizzamia) [inline README.md:407]: Let's add some wording to explain what's going on

File: package.json [hunk 4]
```diff
@@ -18,6 +18,8 @@
     "release:version": "changeset version && yarn install --immutable"
   },
   "peerDependencies": {
+    "graphql": "^16.8.1",
```
Comment (Zizzamia) [inline package.json:21]: Will this clash with any other GraphQL dependcy in the industry? I am a bit ingronant here, but those work this well with Relay?

File: src/core/attestation.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { base } from 'viem/chains';
+import { AttestationSchema } from './types';
+import type { Chain } from 'viem';
+
+export const easScanGraphQLAPI = 'https://base.easscan.org/graphql';
+
```
Comment (Zizzamia) [inline src/core/attestation.ts:6]: Let's split this file in multiple files, where the main one from `core` is the name of the function exported, and anything else goes into utils or store.

File: src/core/getAttestation.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import type { Address } from 'viem';
+import type { Chain } from 'viem';
+import { gql, request } from 'graphql-request';
+import { Attestation, AttestationSchema } from './types';
+import { getChainSchemasUids, easScanGraphQLAPI, getAttesterAddresses } from './attestation';
+
+const attestationsQuery = gql`
```
Comment (Zizzamia) [inline src/core/getAttestation.ts:7]: Would love we have strong types on what's returing and what each of them means.

File: src/core/types.ts [hunk 7]
```diff
@@ -97,3 +97,21 @@ export type FrameMetadataType = {
...
+export type Attestation = {
+  attester: string;
+  expirationTime: number;
+  id: string;
+  recipient: string;
+  revocationTime: number;
+  schemaId: string;
+  timeCreated: number;
+  txid: string;
+};
```
Comment (Zizzamia) [inline src/core/types.ts:115]: Can we add a line of comment for each property, so we know what they are used for.

File: src/core/hasVerifiedAttestations.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ * @param chain - The blockchain to check for attestations.
+ * @param address - The address to check for attestations.
+ * @param expectedSchemas - An array of attestation schemas that are expected.
+ * @returns A promise that resolves to a boolean indicating whether the address has the expected attestations.
+ * @throws Will throw an error if the chain is not supported.
+ */
+export async function hasVerifiedAttestations(
+  chain: Chain,
+  address: Address,
+  expectedSchemas: AttestationSchema[] = [],
```
Comment (Zizzamia) [inline src/core/hasVerifiedAttestations.ts:19]: `expectedSchemas` sounds odd. What exactly are we passing here?

Comment (Zizzamia) [reply src/core/hasVerifiedAttestations.ts:19]: Oh I see we have those `'VERIFIED ACCOUNT' | 'VERIFIED COUNTRY'`, not sure I follow. 

Step 1, tell me the chain
Step 2, tell me an address, ok but who address? Customer address? unclear.
Step 3, tell me a schema, ok what schema? how do I know where those schemas are?

File: src/core/getAttestation.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * Retrieves attestations for a given address and chain, optionally filtered by schemas.
+ *
+ * @param chain - The blockchain of interest.
+ * @param address - The address for which attestations are being queried.
+ * @param filters - Optional filters including schemas to further refine the query.
+ * @returns A promise that resolves to an array of Attestations.
+ * @throws Will throw an error if the request to the GraphQL API fails.
+ */
+export async function getAttestation(
```
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/core/getAttestation.ts:36]: What do we want to codegen? I am generally against codegen as they are the eveil of good developer experience for OSS libraries.

File: package.json (hunk 4 above)
Comment (alvaroraminelli) [reply package.json:21]: wondering if we can say 

```
"graphql": "*",
```

or choose a minium version.

File: src/core/getAttestation.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (Zizzamia) [reply src/core/getAttestation.ts:36]: What kind of codegen we want? Like one for each Attestation? help me understand

File: README.md (hunk 2 above)
Comment (Zizzamia) [reply README.md:412]: ok love that!

File: package.json (hunk 4 above)
Comment (Zizzamia) [reply package.json:21]: Not an expert here, who should we loop in on this?

File: src/core/hasVerifiedAttestations.ts (hunk 8 above)
Comment (Zizzamia) [reply src/core/hasVerifiedAttestations.ts:19]: `address` is probably fine, but we need good example in the docs on how this works and how and when should be used.

File: src/core/getAttestation.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+ * @param chain - The blockchain of interest.
+ * @param address - The address for which attestations are being queried.
+ * @param filters - Optional filters including schemas to further refine the query.
+ * @returns A promise that resolves to an array of Attestations.
+ * @throws Will throw an error if the request to the GraphQL API fails.
+ */
+export async function getAttestation<TChain extends Chain>(
+  address: Address,
+  chain: TChain,
+  filters?: { schemas?: AttestationSchema[] },
```
Comment (Zizzamia) [inline src/core/getAttestation.ts:44]: @cbfyi curios, what other kind of filters we imagine to have later on?

Comment (Zizzamia) [reply src/core/getAttestation.ts:44]: I guess GraphQL filters maybe
```
      where: { AND: [conditions] },
      orderBy: [{ timeCreated: 'desc' }],
      distinct: ['schemaId', 'attester'],
      take: 10,
```

mmmm

**Comment by alvaroraminelli** [comment]
closing as going to break down in smaller PRs.

**Pull request closed without merging**